go/vt/vtgate/safe_session.go

```diff
@@ -16,7 +16,7 @@ import (
 // (the use pattern is 'Find', if not found, then 'Append',
 // for a single shard)
 type SafeSession struct {
-	mu sync.Mutex
+	mu sync.RWMutex
```

Comment:
This one seems useless, have you seen it contribute to tail latency?

(this structure is per-RPC-call, the only concurrent access we have is if multiple go routines inside a single query try to take the lock, and then tail latency is not an issue)